dsc/src/subcommand.rs

```diff
@@ -220,33 +220,45 @@ pub fn config_export(configurator: &mut Configurator, format: &Option<OutputForm
 }
 
 fn initialize_config_root(path: &Option<String>) -> Option<String> {
-    if path.is_some() {
-        let config_path = path.clone().unwrap_or_default();
-        Some(set_dscconfigroot(&config_path))
-    } else if std::env::var(DSC_CONFIG_ROOT).is_ok() {
+    let use_stdin = if let Some(specified_path) = path {
+        if specified_path != "-" {
```

Comment:
I'm trying to understand why `set_dscconfigroot` and input from STDIN are mutually exclusive...
What are the things that we can read from STDIN ? Only configuration itself?

Reply:
I'll add a comment here.  What happens here is that `--file` gets passed all the way to this function so `path` has either `None`, `-`, or an actual path (directory).  So we have to check if `-` is passed which means the path is not valid and we treat this function the same as if no path is provided.  However, the way this function is used, the path is passed back out of it and used later so we have to put back the `-` so subsequent code knows it's STDIN.